### ISSUE TITLE
support multiple Service transports in a didDocs.

### ISSUE THREAD

**Issue by burdettadam** [hyperledger/aries-framework-javascript#383]
After creating an Agent we can set the outbound transporter which will provide the sendmessage method. 
```ts
await this.outboundTransporter.sendMessage(outboundPackage)
```
This is the only transport that will be used and the transporter does not handle the packing, so any attempt at transporter useing multiple services come dependant on all services using the same keys.
Currently, the message sender will attempt to pack and send a message to every service in a didDoc that has an endpoint until one is successful.
To support multiple services it would be nice to be able to:
- register multiple transporters
- update a didDocs service priority
- have the message sender pick the highest supported priority service and use the registered transporter that supports the transport. 

scenario: I have a mediator connection invite I would like to connect to and request mediation from. I use HTTP with return routing to establish the connection. Once the connection is established, the connection didDoc has a service for WebSockets, since I don't have an endpoint I would like to use the WebSocket service for further correspondence with the mediator. While I want to use web socks with the mediated connection, I still want to use HTTP for all other connections since that is the most supported transport in my community.

**Comment by burdettadam** [comment]
duplicate of #268.

**Comment by TimoGlastra** [comment]
Fixed in #419
